tests/Integration/Behaviour/Features/Context/Domain/SupplierFeatureContext.php

```diff
@@ -33,13 +33,74 @@
 use PrestaShop\PrestaShop\Core\Domain\Supplier\Command\AddSupplierCommand;
 use PrestaShop\PrestaShop\Core\Domain\Supplier\Exception\SupplierException;
 use PrestaShop\PrestaShop\Core\Domain\Supplier\Query\GetSupplierForEditing;
+use PrestaShop\PrestaShop\Core\Domain\Supplier\Query\GetSupplierForViewing;
 use PrestaShop\PrestaShop\Core\Domain\Supplier\QueryResult\EditableSupplier;
+use PrestaShop\PrestaShop\Core\Domain\Supplier\QueryResult\ViewableSupplier;
 use PrestaShop\PrestaShop\Core\Domain\Supplier\ValueObject\SupplierId;
+use RuntimeException;
 use State;
+use Supplier;
 use Tests\Integration\Behaviour\Features\Context\Util\PrimitiveUtils;
 
 class SupplierFeatureContext extends AbstractDomainFeatureContext
 {
+    /**
+     * @Given /^supplier "(.+)" with name "(.+)" exists$/
+     *
+     * @param string $reference
+     * @param string $supplierName
+     */
+    public function existsByName(string $reference, string $supplierName): void
```

Comment:
Please move it out of this Domain class, because this method doesn't use CQRS calls.

Reply:
wasn't sure such convention exists, I always though we have just outdated architecture.